### ISSUE TITLE
Rubocop "-a" changes the meaning of the ternary condition

### ISSUE THREAD

**Issue by execat** [rubocop-hq/rubocop#5771]
The rewritten line on executing using "-a" removes the parenthesis which changes the meaning of the condition.

--------

## Expected behavior

`lol.rb`:
```
result = [1, 2, 3].include?(1) ? 'omg' : 'lol'
OR
result = ([1, 2, 3].include? 1) ? 'omg' : 'lol'
puts result
```
The result of this evaluation is "omg".

## Actual behavior

`lol.rb`:
```
result = [1, 2, 3].include? 1 ? 'omg' : 'lol'
puts result
```
The result of this evaluation is 'false'.

## Steps to reproduce the problem

1. Save this as `lol.rb`:
```
result = ([1, 2, 3].include? 1) ? 'omg' : 'lol'
puts result
```

2. Use this `rubocop.yml`:
```
Style/Documentation:
  Enabled: false

Style/StringLiterals:
  Enabled: false

Style/ClassAndModuleChildren:
  Enabled: false

Style/FrozenStringLiteralComment:
  Enabled: false

Metrics/LineLength:
  Max: 120

Metrics/MethodLength:
  Max: 20

Metrics/ClassLength:
  Max: 200

Metrics/ModuleLength:
  Max: 200

Layout/AlignParameters:
  Enabled: false

Metrics/AbcSize:
  Enabled: false

AllCops:
  TargetRubyVersion: 2.3+
  Exclude:
    - '**/*.gemspec'
    - '**/Rakefile'
    - 'lib/sequel/adapters/mariadb.rb'
    - 'Guardfile'

Style/TrailingCommaInArrayLiteral:
  EnforcedStyleForMultiline: comma

Style/TrailingCommaInHashLiteral:
  EnforcedStyleForMultiline: comma

Style/MutableConstant:
  Enabled: false
```

3. Run `rubocop -a lol.rb`

## RuboCop version

```
$ rubocop -V
0.54.0 (using Parser 2.5.1.0, running on ruby 2.4.0 x86_64-darwin16)
```
(But I had this issue on jruby too)

**Comment by Drenmi** [comment]
Hm. I fixed this once, in autumn 2016, but changes have been made to the cop since, so I guess it was inadvertently re-introduced.

**Comment by svendittmer** [comment]
I've started working on this issue, and made some progress fixing it, but the changes affect the behavior of `Style/TernaryParentheses`. So before I continue, I'd like to know if that's acceptable. Currently, `Style/TernaryParentheses` offenses are overlapping with `Style/RedundantParentheses`. Here's two examples where both cops find an offense:
```ruby
foo = (bar?) ? a : b
foo = (bar.baz?) ? a : b
```
On my branch, these offenses are no longer detected by `Style/TernaryParentheses`. Instead, it leaves finding and correcting them to `Style/RedundantParentheses`. So both examples remain offenses and are still autocorrectable.

Is that ok? Or should I attempt to preserve current behavior in the bugfix?

**Comment by koic** [comment]
@svendittmer I think parentheses are unnecessary for method calls without arguments. The following behaviors are the same.

```ruby
% ruby -e "p (nil.nil?) ? 'hi' : 'bye'" # => "hi"
% ruby -e "p nil.nil? ? 'hi' : 'bye'" # => "hi"
```

In these cases without parentheses, warnings are not registered with `Style/TernaryParentheses` cop and `Style/RedundantParentheses ` cop.

**Comment by koic** [comment]
This issue duplicates #4460.
Please note the test case of https://github.com/rubocop-hq/rubocop/issues/4460#issuecomment-362282999. This case is not a method call but a conditional expression.

**Comment by Drenmi** [comment]
Closing as duplicate. Please follow #4460 for updates.
